fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -821,19 +819,9 @@ public FHIRRestOperationResponse doPatchOrUpdatePersist(FHIRPersistenceEvent eve
             } else {
                 // prevResource exists
                 if (result.getStatus() == InteractionStatus.IF_NONE_MATCH_EXISTED) {
-                    // Use the location assigned to the previous resource because we're
-                    // not updating anything. Also, we don't fire any 'after' event
-                    // for the same reason.
-                    ior.setResource(prevResource); // 304 Not Modified never needs to return content
-                    ior.setLocationURI(FHIRUtil.buildLocationURI(type, id, result.getIfNoneMatchVersion()));
-                    Boolean ifNoneMatchNotModified = FHIRConfigHelper.getBooleanProperty(
-                        FHIRConfiguration.PROPERTY_IF_NONE_MATCH_RETURNS_NOT_MODIFIED, Boolean.FALSE);
-                    if (ifNoneMatchNotModified != null && ifNoneMatchNotModified) {
-                        // Don't treat as an error
-                        ior.setStatus(Response.Status.NOT_MODIFIED);
-                    } else {
-                        throw new FHIRPersistenceIfNoneMatchException("IfNoneMatch precondition failed.");
-                    }
+                    // this should have been handled before the db interaction; this is just a double check
```

Comment:
It's to avoid a race condition - because this check was integrated into the SELECT FOR UPDATE locking we do, which means we correctly handle the if-none-match case where a resource has been deleted and two threads are trying to create it again.

Reply:
for sure its needed in the case above (`if (createOnUpdate)`), but here we are in the else block which means that we found a resource when we did the read, which means we should have caught it when we called doUpdateMeta

Reply:
OK. I didn't get the full context from the diff...I just thought a fuller comment would be helpful